app/actions/remote/post.ts

```diff
@@ -182,6 +188,75 @@ export async function createPost(serverUrl: string, post: Partial<Post>, files:
     return {data: true};
 }
 
+export const retryFailedPost = async (serverUrl: string, post: PostModel) => {
+    const operator = DatabaseManager.serverDatabases[serverUrl]?.operator;
+    if (!operator) {
+        return {error: `${serverUrl} database not found`};
+    }
+
+    let client: Client;
+    try {
+        client = NetworkManager.getClient(serverUrl);
+    } catch (error) {
+        return {error};
+    }
+
+    try {
+        const timestamp = Date.now();
+        const apiPost = await post.toApi();
+        const newPost = {
+            ...apiPost,
+            props: {
+                ...apiPost.props,
+                failed: false,
+            },
+            id: '',
+            create_at: timestamp,
+            update_at: timestamp,
+            delete_at: 0,
+        } as Post;
+
+        // Update the local post to reflect the pending state in the UI
+        // timestamps will remain the same as the initial attempt
+        post.prepareUpdate((p) => {
```

Comment:
If the timestamp remains as the initial attempt, the "Time to fail" logic we have will trigger again, and will look as failed from the beginning. Either we change the "time to fail" logic, or the timestamp.